sdks/java/core/src/main/java/org/apache/beam/sdk/util/VarInt.java

```diff
@@ -42,13 +42,33 @@ public static void encode(int v, OutputStream stream) throws IOException {
 
   /** Encodes the given value onto the stream. */
   public static void encode(long v, OutputStream stream) throws IOException {
+    // Write 1-5 bytes
+    if ((v & ~0xFFFFFFFFL) == 0) {
```

Comment:
This is mostly a remnant from when I was thinking this should be split between ints and longs.
Currently it will split on the middle of the 5th continuation byte, anything using the top 3 bits (longs) of that byte will write to the byte array, anything using the lower 4 bits (ints) will write directly to the stream.

This could be any mask and might be worth changing based on the x86-64 results such that anything which produces two bytes writes directly to the stream and the rest to the byte array.